native_posix: 💧 Fix realloc potential leak

noahp opened the pull request:
Fix a possible leak if `realloc` fails here; check the result of the
`realloc` call before updating the pointer, so it can be freed in the
failure case.

andrewboie commented:
heh, does our coding standard allow emojis in commit messages?

tyrantlizard77 reviewed (approved):
code LGTM

aescolar reviewed (changes requested):
Please remove the emoji from the commit title, and use a real email address in the commit author and signed-off-by message.
The code change is ok otherwise.

noahp commented:
:grinning: yeah, I didn't see that prohibited anywhere in the coding standard, happy to remove it!

aescolar commented:
@noahp You will need to amend the commit author to match the signed off address

noahp commented:
> @noahp You will need to amend the commit author to match the signed off address

Oof yeah, my b, sorry about that.

noahp committed:
native_posix: Fix realloc potential leak

Fix a possible leak if `realloc` fails here; check the result of the
`realloc` call before updating the pointer, so it can be freed in the
failure case.

Signed-off-by: Noah Pendleton <noah.pendleton@gmail.com>

aescolar reviewed (approved):
Thanks
